### PR TITLE
chore: Document options naming and remediate

### PR DESCRIPTION
**Pull request by jamtur01**
Example changes in this PR:

* Fingerprinting becomes Fingerprint.
* The `fingerprint_bytes` option becomes `bytes`.
* The `merge_fields` option becomes `fields`.

### CODE REVIEW

File: config/vector.spec.toml [hunk 1]
```diff
@@ -258,7 +258,7 @@ data_dir = "/home/user/vector/"
   # Fingerprinting
   #
 
-  [sources.file.fingerprinting]
+  [sources.file.fingerprinti]
```
Comment (MOZGIII) [inline config/vector.spec.toml:261]:
```suggestion
  [sources.file.fingerprint]
```

Comment (jamtur01) [reply config/vector.spec.toml]: Thanks.

**Review by MOZGIII — APPROVED** [review]
It's really nice to have some guidelines on naming!

**Comment by binarylogic** [comment]
The name changes look great, I would just like to think about backward compatibility. I assume it would be easy to map the old names to the new names?

**Comment by jamtur01** [comment]
@binarylogic I've added aliases to the `fingerprinting` and `merge_fields` fields but I am unsure who to exactly handle deprecation here. @lukesteensen Do you have input?

**Review by lukesteensen — APPROVED** [review]
You've done the aliasing correctly, but we don't necessarily have a good way to surface the deprecation to the user right now. That'd require something a bit fancier to plug into deserialization. I'll make a note to consider this in the config RFC.

**Comment by jamtur01** [comment]
Thanks!

File: src/sources/file.rs [hunk 2]
```diff
@@ -80,26 +82,26 @@ pub struct FileConfig {
 
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
 #[serde(tag = "strategy", rename_all = "snake_case")]
-pub enum FingerprintingConfig {
+pub enum FingerprintConfig {
     Checksum {
-        fingerprint_bytes: usize,
```
Comment (juchiast) [inline src/sources/file.rs:85]: This is a breaking change, should we add serde alias here too?

Comment (binarylogic) [reply src/sources/file.rs:85]: Yes, we should. Would you mind opening a PR for that? I don't think James or myself know how to do that.

Comment (juchiast) [reply src/sources/file.rs:85]: Submitted!

**Pull request merged**